### PR TITLE
`d/azurerm_kubernetes_cluster`: Support `local_account_disabled` setup in data source

### LINKED ISSUE #13248: Support for Local Account disabled for AKS
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

To upgrade security for AKS significantly, we would like to have [the option to disable local accounts](https://docs.microsoft.com/en-us/azure/aks/managed-aad#disable-local-accounts-preview) implemented.

When this would be implemented, you are unable to get a working kubeconfig without AAD login. This removes the threat of misuse of the local kubeconfig file of an administrator.

Using the Kubernetes provider, kubectl provider or Helm provider would be done as shown below:

The configuration as shown below does not work when managed AAD is enabled, requiring a little fix.

### New or Affected Resource(s)

- [x] azurerm_kubernetes_cluster, resource (#12386)
- [ ] azurerm_kubernetes_cluster, data source (#13260)

### Potential Terraform Configuration

```hcl
# infra/main.tf
resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "West Europe"
}

resource "azurerm_kubernetes_cluster" "example" {
  name                   = "example-aks1"
  location               = azurerm_resource_group.example.location
  resource_group_name    = azurerm_resource_group.example.name
  dns_prefix             = "exampleaks1"
  local_account_disabled = true

  default_node_pool {
    name       = "default"
    node_count = 1
    vm_size    = "Standard_D2_v2"
  }

  role_based_access_control {
    enabled                 = true

    azure_active_directory {
      managed                = true
      admin_group_object_ids = var.cluster_admin_groups
    }
  }

  identity {
    type = "SystemAssigned"
  }

  tags = {
    Environment = "Production"
  }
}
```

```hcl
# application/main.tf
data "azurerm_kubernetes_cluster" "default" {
  name                = "myakscluster"
  resource_group_name = "my-example-resource-group"
}


data "azurerm_kubernetes_cluster" "default" {
  name                = "${var.prefix}-aks"
  resource_group_name = azurerm_resource_group.example.name
}

data "azurerm_client_config" "current" {}

provider "kubernetes" {
  host                   = data.azurerm_kubernetes_cluster.default.kube_config.0.host
  cluster_ca_certificate = base64decode(data.azurerm_kubernetes_cluster.default.kube_config.0.cluster_ca_certificate)

  exec {
    api_version = "client.authentication.k8s.io/v1beta1"
    command     = "kubelogin"
    args = [
      "get-token",
      "--login",
      "azurecli", # spn if you want to use service principal, but requires use of env vars AAD_SERVICE_PRINCIPAL_CLIENT_ID and AAD_SERVICE_PRINCIPAL_CLIENT_SECRET
      "--environment",
      "AzurePublicCloud",
      "--tenant-id",
      data.azurerm_client_config.current.tenant_id,
      "--server-id",
      "6dae42f8-4368-4678-94ff-3960e28e3630",
      "|",
      "jq",
      ".status.token"
    ]
  }
}

provider "helm" {
  kubernetes {
    host                   = data.azurerm_kubernetes_cluster.default.kube_config.0.host
    cluster_ca_certificate = base64decode(data.azurerm_kubernetes_cluster.default.kube_config.0.cluster_ca_certificate)

    exec {
      api_version = "client.authentication.k8s.io/v1beta1"
      command     = "kubelogin"
      args = [
        "get-token",
        "--login",
        "azurecli", # spn if you want to use service principal, but requires use of env vars AAD_SERVICE_PRINCIPAL_CLIENT_ID and AAD_SERVICE_PRINCIPAL_CLIENT_SECRET
        "--environment",
        "AzurePublicCloud",
        "--tenant-id",
        data.azurerm_client_config.current.tenant_id,
        "--server-id",
        "6dae42f8-4368-4678-94ff-3960e28e3630",
        "|",
        "jq",
        ".status.token"
      ]
    }
  }
}
```

### References

* #13168
* https://github.com/hashicorp/terraform-provider-kubernetes/issues/1176

### PR DESCRIPTION
**Pull request by aristosvo**
Fixes #13248

## Acceptance Test
- [x] Test added for fixed/new functionality: TestAccDataSourceKubernetesCluster_localAccountDisabled
- [x] New test run:
```bash
❯ make acctests SERVICE='containers' TESTARGS='-run=TestAccDataSourceKubernetesCluster_localAccountDisabled'
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./internal/services/containers -run=TestAccDataSourceKubernetesCluster_localAccountDisabled -timeout 180m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccDataSourceKubernetesCluster_localAccountDisabled
=== PAUSE TestAccDataSourceKubernetesCluster_localAccountDisabled
=== CONT  TestAccDataSourceKubernetesCluster_localAccountDisabled
--- PASS: TestAccDataSourceKubernetesCluster_localAccountDisabled (473.35s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/containers    474.821s
```

### CODE REVIEW

File: website/docs/d/kubernetes_cluster.html.markdown [hunk 1]
```diff
@@ -52,9 +52,9 @@ The following attributes are exported:
 
 -> **NOTE:**  At this time Private Link is in Public Preview.
 
-* `kube_admin_config` - A `kube_admin_config` block as defined below. This is only available when Role Based Access Control with Azure Active Directory is enabled.
+* `kube_admin_config` - A `kube_admin_config` block as defined below. This is only available when Role Based Access Control with Azure Active Directory is enabled and local accounts not disabled.
```
Comment (katbyte) [inline website/docs/d/kubernetes_cluster.html.markdown:55]:
```suggestion
* `kube_admin_config` - A `kube_admin_config` block as defined below. This is only available when Role Based Access Control with Azure Active Directory is enabled and local accounts are not disabled.
```

File: website/docs/d/kubernetes_cluster.html.markdown [hunk 2]
```diff
@@ -52,9 +52,9 @@ The following attributes are exported:
...
 
-* `kube_admin_config_raw` - Raw Kubernetes config for the admin account to be used by [kubectl](https://kubernetes.io/docs/reference/kubectl/overview/) and other compatible tools. This is only available when Role Based Access Control with Azure Active Directory is enabled.
+* `kube_admin_config_raw` - Raw Kubernetes config for the admin account to be used by [kubectl](https://kubernetes.io/docs/reference/kubectl/overview/) and other compatible tools. This is only available when Role Based Access Control with Azure Active Directory is enabled and local accounts not disabled.
```
Comment (katbyte) [inline website/docs/d/kubernetes_cluster.html.markdown:57]:
```suggestion
* `kube_admin_config_raw` - Raw Kubernetes config for the admin account to be used by [kubectl](https://kubernetes.io/docs/reference/kubectl/overview/) and other compatible tools. This is only available when Role Based Access Control with Azure Active Directory is enabled and local accounts are not disabled.
```

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @aristosvo - couple minor comments and it looks like the test is failing: 
```
------- Stdout: -------
=== RUN   TestAccDataSourceKubernetesCluster_localAccountDisabled
=== PAUSE TestAccDataSourceKubernetesCluster_localAccountDisabled
=== CONT  TestAccDataSourceKubernetesCluster_localAccountDisabled
    testcase.go:88: Step 1/1 error: Check failed: Check 8/8 error: data.azurerm_kubernetes_cluster.test: Attribute 'kube_admin_config_raw' found when not expected
--- FAIL: TestAccDataSourceKubernetesCluster_localAccountDisabled (532.56s)
FAIL
```

File: website/docs/d/kubernetes_cluster.html.markdown (hunk 1 above)
Comment (aristosvo) [reply website/docs/d/kubernetes_cluster.html.markdown]: Thanks, resolved! 🥳

File: website/docs/d/kubernetes_cluster.html.markdown (hunk 2 above)
Comment (aristosvo) [reply website/docs/d/kubernetes_cluster.html.markdown]: Thanks, fixed! 🎈

**Comment by aristosvo** [comment]
@katbyte Thanks for the quick review! Was already running (and fixing) locally, so I believe it is solved for now 👍🏽

**Review by katbyte — APPROVED** [review]
Thanks @aristosvo! LGTM 🏗️

**Pull request merged**